modules/passkey/contracts/libraries/P256.sol

```diff
@@ -94,4 +130,33 @@ library P256 {
             )
         }
     }
+
+    /**
+     * @notice Verifies the signature of a message using P256 elliptic curve, without signature
+     * malleability check.
+     * @param verifiers The P-256 verifiers to use.
+     * @param message The signed message.
+     * @param r The r component of the signature.
+     * @param s The s component of the signature.
+     * @param x The x coordinate of the public key.
+     * @param y The y coordinate of the public key.
+     * @return success A boolean indicating whether the signature is valid or not.
+     */
+    function verifySignatureAllowMalleability(
+        Verifiers verifiers,
+        bytes32 message,
+        uint256 r,
+        uint256 s,
+        uint256 x,
+        uint256 y
+    ) internal view returns (bool success) {
+        address precompileVerifier = address(uint160(uint256(Verifiers.unwrap(verifiers)) >> 160));
+        address fallbackVerifier = address(uint160(Verifiers.unwrap(verifiers)));
+        if (precompileVerifier != address(0)) {
+            success = verifySignatureAllowMalleability(IP256Verifier(precompileVerifier), message, r, s, x, y);
+        }
+        if (!success && fallbackVerifier != address(0)) {
```

Comment:
To confirm: Won't this check the signature twice in case of a wrong signature even if the precompile is now available?

Though don't think it could be a problem, as in case of signature being correct, it only checks once 👍🏾

Reply:
Yes, this is correct. I will clarify why in a comment, but basically, it is because the P-256 precompile does not differentiate between invalid signature and not being available (notably, in both cases the empty bytes `bytes("")` are returned).